### PR TITLE
Cache Image configs by name

### PR DESCRIPTION
**Pull request by dgageot**
This makes it possible to redeploy code faster by not having to lookup for an image’s config if we already fetched it.

Signed-off-by: David Gageot <david@gageot.net>

### CODE REVIEW

File: pkg/skaffold/docker/parse.go [hunk 1]
```diff
@@ -123,7 +121,14 @@ func processBaseImage(value *parser.Node) ([]string, error) {
 	return cfg.Config.OnBuild, nil
 }
 
+var imageToConfigCache sync.Map
+
 func retrieveImageConfig(image string) (*manifest.Schema2Image, error) {
+	cachedCfg, present := imageToConfigCache.Load(image)
```
Comment (r2d4) [inline pkg/skaffold/docker/parse.go]: What if the image gets overwritten in the registry?

:latest, someone clobbering existing tag, etc.

Comment (dlorenc) [reply pkg/skaffold/docker/parse.go]: Yeah I thought about that a bit, this kind of matches docker build semantics though. It won't pull an image for updates if it's already pulled locally unless you force it to.

Comment (dgageot) [reply pkg/skaffold/docker/parse.go]: Exactly

Comment (r2d4) [reply pkg/skaffold/docker/parse.go]: That makes sense, LGTM!

**Pull request merged**